tests/golang/basic_step_test.go

```diff
@@ -153,117 +153,110 @@ func TestFunctionSteps(t *testing.T) {
 	})
 
 	t.Run("trace run should have appropriate data", func(t *testing.T) {
-		<-time.After(3 * time.Second)
-
-		require.Eventually(t, func() bool {
-			run := c.RunTraces(ctx, runID)
-			require.NotNil(t, run)
-			require.Equal(t, models.FunctionStatusCompleted.String(), run.Status)
-			require.False(t, run.IsBatch)
-			require.Nil(t, run.BatchCreatedAt)
-
-			require.NotNil(t, run.Trace)
-			require.True(t, run.Trace.IsRoot)
-			require.Equal(t, 5, len(run.Trace.ChildSpans))
-			require.Equal(t, models.RunTraceSpanStatusCompleted.String(), run.Trace.Status)
-
+		run := c.WaitForRunTraces(ctx, t, &runID, models.FunctionStatusCompleted)
```

Comment:
All that's changed about all affected tests is that we move away from the previous flow of
- Waiting an arbitrary duration which causes flakiness (`<-time.After(3 * time.Second)`)
- Improperly using require which causes panics when flakiness is encountered (`require.NotNil(t, run)`)
- Switch away from `require.EventuallyWithT(t, func(ct *assert.CollectT) {` to using the helper to prevent running into accidental assertion panics